src/modals/custom-rpc-modals/add-modal.tsx

```diff
@@ -0,0 +1,72 @@
+import { Formik } from 'formik';
+import React, { FC } from 'react';
+import { View } from 'react-native';
+import { useDispatch } from 'react-redux';
+
+import { ButtonLargePrimary } from 'src/components/button/button-large/button-large-primary/button-large-primary';
+import { ButtonLargeSecondary } from 'src/components/button/button-large/button-large-secondary/button-large-secondary';
+import { ButtonsContainer } from 'src/components/button/buttons-container/buttons-container';
+import { Divider } from 'src/components/divider/divider';
+import { InsetSubstitute } from 'src/components/inset-substitute/inset-substitute';
+import { Label } from 'src/components/label/label';
+import { ModalStatusBar } from 'src/components/modal-status-bar/modal-status-bar';
+import { ScreenContainer } from 'src/components/screen-container/screen-container';
+import { FormTextInput } from 'src/form/form-text-input';
+import { RpcInterface } from 'src/interfaces/rpc.interface';
+import { ModalsEnum } from 'src/navigator/enums/modals.enum';
+import { useNavigation } from 'src/navigator/hooks/use-navigation.hook';
+import { addCustomRpc, setSelectedRpcUrl } from 'src/store/settings/settings-actions';
+import { useRpcListSelector } from 'src/store/settings/settings-selectors';
+import { formatSize } from 'src/styles/format-size';
+import { showErrorToast } from 'src/toast/toast.utils';
+import { usePageAnalytic } from 'src/utils/analytics/use-analytics.hook';
+
+import { formInitialValues, formValidationSchema } from './form.utils';
+
+export const AddCustomRpcModal: FC = () => {
+  const dispatch = useDispatch();
+  const { goBack } = useNavigation();
+  const rpcList = useRpcListSelector();
+
+  const handleSubmit = (newRpc: RpcInterface) => {
+    const duplicate = rpcList.find(rpc => rpc.name === newRpc.name || rpc.url === newRpc.url);
+
+    if (duplicate != null) {
```

Comment:
use `isDefined`

Reply:
it's very safe and comfortable that this function covers `undefined` and `null` cases and works as a type guard.  
you don't need to worry that something except "defined" type will pass trough